### PR TITLE
Add a Node

### PR DESCRIPTION
**Pull request by HCastano**
Adds a node to use for testing out our bridge runtimes. It'll first be used by #21.

I have all of the Substrate dependencies pinned to `v2.0.0-alpha.2` at specific commit. I'm okay getting this merged in like that and updating it once everything ends up on crates.io.

### CODE REVIEW

**Comment by HCastano** [comment]
I need to get the WASM toolchain in the CI to make Travis happy, but I'll do that tomorrow(ish).

File: bin/node/node/src/chain_spec.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	AccountId, AuraConfig, BalancesConfig, GenesisConfig, GrandpaConfig,
+	SudoConfig, SystemConfig, WASM_BINARY, Signature
+};
+use sp_consensus_aura::sr25519::{AuthorityId as AuraId};
+use grandpa_primitives::{AuthorityId as GrandpaId};
+use sc_service;
+use sp_runtime::traits::{Verify, IdentifyAccount};
+
+// Note this is the URL for the telemetry server
+//const STAGING_TELEMETRY_URL: &str = "wss://telemetry.polkadot.io/submit/";
```
Comment (tomusdrw) [inline bin/node/node/src/chain_spec.rs]: Remove?

File: bin/node/node/src/chain_spec.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+/// The chain specification option. This is expected to come in from the CLI and
+/// is little more than one of a number of alternatives which can easily be converted
+/// from a string (`--chain=...`) into a `ChainSpec`.
+#[derive(Clone, Debug)]
+pub enum Alternative {
+	/// Whatever the current runtime is, with just Alice as an auth.
+	Development,
+	/// Whatever the current runtime is, with simple Alice/Bob auths.
+	LocalTestnet,
```
Comment (tomusdrw) [inline bin/node/node/src/chain_spec.rs]: Do we really need both initially?

File: bin/node/runtime/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,353 @@
+//! The Substrate Node Template runtime. This can be compiled with `#[no_std]`, ready for Wasm.
```
Comment (tomusdrw) [inline bin/node/runtime/src/lib.rs]: Could we extend `polkadot/kusama` runtime instead? Perhaps use common stuff from their repo?

One thing that might not work is that we will end up in git dependency hell if we don't solve #20 first.

Comment (HCastano) [reply bin/node/runtime/src/lib.rs]: Is there anything in particular that you'd like to get out from their runtime?

File: bin/node/node/src/chain_spec.rs (hunk 2 above)
Comment (HCastano) [reply bin/node/node/src/chain_spec.rs]: Guess not, I'll remove the `LocalTestnet` variant

File: bin/node/runtime/src/lib.rs (hunk 3 above)
Comment (HCastano) [reply bin/node/runtime/src/lib.rs]: Talked about this off Github. What we'll do is keep the template for now, and once we have made more substantial progress with the runtimes we'll add a Kusama/Polkadot node for testing.

**Pull request merged**